packages/frontend/src/ee/pages/AiAgents/AgentThreadPage.tsx

```diff
@@ -0,0 +1,307 @@
+                        <Loader />
+                    ) : vizQuery.isError ? (
+                        <Text>Error fetching viz</Text>
+                    ) : vizQuery.data.type === 'vertical_bar_chart' ? (
+                        <EChartsReact option={vizQuery.data.chartOptions} />
+                    ) : vizQuery.data.type === 'time_series_chart' ? (
+                        <EChartsReact option={vizQuery.data.chartOptions} />
+                    ) : vizQuery.data.type === 'csv' ? (
+                        <AiTableViz results={vizQuery.data.results} />
+                    ) : (
+                        assertUnreachable(
+                            vizQuery.data.type,
+                            'Unknown viz type',
+                        )
+                    )}
+                </Paper>
+            )}
+            <Group gap={0}>
+                <CopyButton value={message.message}>
+                    {({ copied, copy }) => (
+                        <ActionIcon
+                            variant="subtle"
+                            color="gray"
+                            aria-label="copy"
+                            onClick={copy}
+                            style={{ display: isLoading ? 'none' : 'block' }}
+                        >
+                            {copied ? (
+                                <IconCheck size={16} />
+                            ) : (
+                                <IconCopy size={16} />
+                            )}
+                        </ActionIcon>
+                    )}
+                </CopyButton>
+                {/* TODO: Add up/down vote to Web UI */}
+                <Group style={{ display: 'none' }}>
+                    <ActionIcon
+                        variant="subtle"
+                        color={upVoted ? 'green' : 'gray'}
+                        aria-label="upvote"
+                    >
+                        <IconThumbUp size={16} />
+                    </ActionIcon>
+
+                    <ActionIcon
+                        variant="subtle"
+                        color={downVoted ? 'red' : 'gray'}
+                        aria-label="downvote"
+                    >
+                        <IconThumbDown size={16} />
+                    </ActionIcon>
+                </Group>
+            </Group>
+        </Stack>
+    );
+};
+
+type AiThreadMessageProps = {
+    message: AiAgentMessage;
+    agentName?: string;
+};
+
+const AiThreadMessage: FC<AiThreadMessageProps> = ({ message }) => {
+    const isUser = message.role === 'user';
+
+    return isUser ? (
+        <UserBubble message={message} />
+    ) : (
+        <AssistantBubble message={message} />
+    );
+};
+
+const shouldRenderDivider = (messages: AiAgentMessage[], idx: number) => {
+    if (idx === 0) return true;
+    const prev = messages[idx - 1];
+    const curr = messages[idx];
+    return !dayjs(prev.createdAt).isSame(curr.createdAt, 'day');
+};
```

Comment:
[sugg] do we need `Today`? I think we can also filter that one out

![CleanShot 2025-06-03 at 12.28.08@2x.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/SRLEqMEevAzoFwvhnfeq/4194bf6b-3454-43fd-a83a-712014a97605.png)